### PR TITLE
More deterministic move accuracy validation.

### PR DESCRIPTION
**Pull request by Sopel97**
This PR does a 'manual' depth 1 search instead of the stateful `search(pos, 1)` where qsearch and move ordering matters. Also require the best move to have a strictly higher score than the next one.

### CODE REVIEW

**Comment by nodchip** [comment]
@Sopel97 How are move accuracies changed in the training log?  Could  you please show me differences?  Also, does this slow down the training?

**Comment by Sopel97** [comment]
This should be faster if anything because there's no need to initialize the search and depth 1 search looks through at least all legal moves anyway. Though the difference is negligible.

Example logs (speeds not comparable because machine busy):
```
setoption name SkipLoadingEval value true
setoption name Threads value 4
setoption name Use NNUE value pure
learn targetdir data epochs 400 batchsize 100000 use_draw_in_training 1 use_draw_in_validation 1 lr 1 lambda 1 eval_limit 9000 max_grad 0.3 nn_batch_size 1000 newbob_decay 0.99 eval_save_interval 10000000 loss_output_interval 1000000 smart_fen_skipping set_recommended_uci_options verbose
```
master: https://pastebin.com/VT6QdfYa
PR: https://pastebin.com/m0MWeFd9

Move accuracy will be generally lower because it's not doing qsearch and there's no history tables. It should be more indicative of strength though and definitely doesn't change when the trainer is reloaded with the same net.

File: src/learn/learn.cpp [hunk 1]
```diff
@@ -971,15 +975,49 @@ namespace Learner
...
+            if (has_depth1_move_agreement(pos, (Move)ps.move))
                 move_accord_count.fetch_add(1, std::memory_order_relaxed);
         }
 
         test_loss_sum += local_loss_sum;
     }
 
+    bool LearnerThink::has_depth1_move_agreement(Position& pos, Move pvmove)
+    {
+        // Determine if the depth 1 search pv matches the move from the dataset.
```
Comment (nodchip) [inline src/learn/learn.cpp:988]: Could we call Eval::evaluate() here to enable incremental computation of eval values?

**Comment by nodchip** [comment]
> This should be faster if anything because there's no need to initialize the search and depth 1 search looks through at least all legal moves anyway. Though the difference is negligible.

It makes sense.  Thank you.

> Example logs (speeds not comparable because machine busy):
The logs look fine to me.  Thank you.

Could you please take a look at a review comment?

File: src/learn/learn.cpp (hunk 1 above)
Comment (Sopel97) [reply src/learn/learn.cpp:988]: good point, done

**Comment by nodchip** [comment]
@Sopel97 Thank you for the fix.  Could you please rebase this PR?

**Comment by Sopel97** [comment]
should be ready

**Pull request merged**